doc/challenge/README.md

```diff
@@ -2,7 +2,22 @@
 This document provides some ideas about possible challenges Anna should be able
 to accomplish at some point.
 
-- Interacting with humans using human language.
-- Image recognition. E.g. recognizing numbers of images.
-- Doing Math.
-- TBD
+### basic
+Basic challenges describe fundamental mechasnisms that needs to be solved to be
+able to solve more sophisticated problems.
+
+- [guidance](basic/guidance.md)
+- [pattern recognition](basic/pattern_recognition.md)
+- planning? what is planning? how can planning work?
+- selective perception
+- context awareness
+
+### advanced
+Advanced challenges describe a very specific problem that needs to be solved.
+It may require one or multiple basic challenges to be accomplished first before
+tackling a advanced challenge.
+
+- [starting a conversation by greeting](advanced/starting_a_conversation_by_greeting.md)
+- detect patterns of word order of sentences
+- detect learning and action/reaction patterns
+- understanding of what a quote is and what the character `"` is good for
```

Comment:
This is way to unstructured. All information in here are only thrown together somehow.